test/metabase/test/data/dataset_definition_test.clj

```diff
@@ -30,7 +30,7 @@
                    :fk_target_field_id nil
                    :semantic_type      :type/PK}]
                  user-fields)))
-        (when-not (#{:sqlite} driver/*driver*) ;; our implement does not support adding fk for custom dataset yet
```

Comment:
Note that SQLite doesn't support FKs in tests, so this change still works for SQLite. See
https://github.com/metabase/metabase/blob/6509a0e8b48326351bd4986a72cd6141bfdb70fd/modules/drivers/sqlite/src/metabase/driver/sqlite.clj#L47-L49